From 27ce3565136f8c31c69e10b234a3885c9abfb8d7 Mon Sep 17 00:00:00 2001
From: Laurentiu Tudor <laurentiu.tudor@nxp.com>
Date: Tue, 11 Dec 2018 13:32:31 +0200
Subject: [PATCH 417/741] arm64: dts: ls104x: constrain sata dma address size

commit 1ce29eb637cbe16422e777af4620a01d7462945b from
http://source.codeaurora.org/external/qoriq/qoriq-components/linux

Limit the dma mask size for sata to 40 bits to match the actual address
size generated towards the interconnect. Re-use the already existing
auxiliary simple bus meant for usb but drop the usb reference from the
node name.

Signed-off-by: Laurentiu Tudor <laurentiu.tudor@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 .../arm64/boot/dts/freescale/fsl-ls1043a.dtsi | 20 +++++++++----------
 1 file changed, 10 insertions(+), 10 deletions(-)

diff --git a/arch/arm64/boot/dts/freescale/fsl-ls1043a.dtsi b/arch/arm64/boot/dts/freescale/fsl-ls1043a.dtsi
index b5c9a6bfd02a..6fc1cffcfcb9 100644
--- a/arch/arm64/boot/dts/freescale/fsl-ls1043a.dtsi
+++ b/arch/arm64/boot/dts/freescale/fsl-ls1043a.dtsi
@@ -694,7 +694,7 @@
 				 <&clockgen 4 0>;
 		};
 
-		usb_aux_bus: usb_aux_bus {
+		aux_bus: aux_bus {
 			#address-cells = <2>;
 			#size-cells = <2>;
 			compatible = "simple-bus";
@@ -739,16 +739,16 @@
 				snps,incr-burst-type-adjustment = <1>, <4>, <8>, <16>;
 				configure-gfladj;
 			};
-		};
 
-		sata: sata@3200000 {
-			compatible = "fsl,ls1043a-ahci";
-			reg = <0x0 0x3200000 0x0 0x10000>,
-				<0x0 0x20140520 0x0 0x4>;
-			reg-names = "ahci", "sata-ecc";
-			interrupts = <0 69 0x4>;
-			clocks = <&clockgen 4 0>;
-			dma-coherent;
+			sata: sata@3200000 {
+				compatible = "fsl,ls1043a-ahci";
+				reg = <0x0 0x3200000 0x0 0x10000>,
+					<0x0 0x20140520 0x0 0x4>;
+				reg-names = "ahci", "sata-ecc";
+				interrupts = <0 69 0x4>;
+				clocks = <&clockgen 4 0>;
+				dma-coherent;
+			};
 		};
 
 		qdma: qdma@8380000 {
-- 
2.17.1

